hummingbot/strategy/avellaneda_market_making/avellaneda_market_making.pyx

```diff
@@ -1249,6 +1286,41 @@ cdef class AvellanedaMarketMakingStrategy(StrategyBase):
         if self._cancel_timestamp <= self._current_timestamp:
             self._cancel_timestamp = min(self._create_timestamp, next_cycle)
 
+    cdef bint c_is_trading_allowed(self):
+        if self._execution_timeframe == "from_date_to_date":
+            # Only within specific dates
+            return self._start_time.timestamp() <= self._current_timestamp <= self._end_time.timestamp()
+        elif self._execution_timeframe == "daily_between_times":
+            # Only within specific times of a day
+            current_time = datetime.datetime.fromtimestamp(self._current_timestamp).time()
+            return self._start_time <= current_time <= self._end_time
+        elif self._execution_timeframe == "infinite":
+            # Infinite - always allowed
+            return True
+        else:
+            # Invalid timeframe value - error
+            raise Exception("Invalid timeframe")
```

Comment:
We should put the execution timeframe logic inside a separate component (similar to what we did for the TWAP trade strategy). And each of the different alternatives for timeframes could be a different subclass in the hierarchy.